cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/CordaTestBase.kt

```diff
@@ -0,0 +1,145 @@
+     * */
+    protected lateinit var net: InternalMockNetwork
+        private set
+
+    private val parties: MutableList<StartedNode<MockNode>> = mutableListOf()
+
+    protected val random = Random()
+
+    /**
+     * Recreate nodes before each test
+     *
+     * Usage:
+     *
+     *     lateinit var issuer: StartedNode<MockNode>
+     *
+     *     @Before
+     *     createNodes() {
+     *         issuer = createPartyNode(CordaX500Name("Issuer", "London", "GB"))
+     *     }
+     * */
+    protected fun createPartyNode(legalName: CordaX500Name? = null): StartedNode<MockNode> {
+        val party = net.createPartyNode(legalName)
+
+        parties.add(party)
+
+        for(flow in projectReciverFlows) {
+            party.registerInitiatedFlow(flow.java)
+        }
+
+        return party
+    }
+
+    /**
+     * Substitutes [StartedNodeServices.startFlow] method to run mocked Corda flows.
+     *
+     * Usage:
+     *
+     *     val did = store.services.startFlow(GetDidFlow.Initiator(name)).resultFuture.get()
+     */
+    protected fun <T> StartedNodeServices.startFlow(logic: FlowLogic<T>): FlowStateMachine<T> {
+        val machine = startFlow(logic, newContext()).getOrThrow()
+
+        return object : FlowStateMachine<T> by machine {
+            override val resultFuture: CordaFuture<T>
+                get() {
+                    net.runNetwork()
+                    return machine.resultFuture
+                }
+        }
+    }
+
+    @Before
+    fun commonSetup() {
+        TestConfigurationsProvider.provider = object : TestConfigurationsProvider {
+            override fun getConfig(name: String): Configuration? {
+                // Watch carefully for these hard-coded values
+                // Now we assume that issuer(indy trustee) is the first created node from SomeNodes
+                return if (name == "Issuer") {
+                    ConfigurationMap(mapOf(
+                            "indyuser.walletName" to name,
+                            "indyuser.role" to "trustee",
+                            "indyuser.did" to "V4SGRU86Z58d6TV7PBUe6f",
+                            "indyuser.seed" to "000000000000000000000000Trustee1",
+                            "indyuser.genesisFile" to PoolManager.DEFAULT_GENESIS_FILE
+                    ))
+                } else ConfigurationMap(mapOf(
+                        "indyuser.walletName" to name + random.nextLong().absoluteValue,
+                        "indyuser.genesisFile" to PoolManager.DEFAULT_GENESIS_FILE
+                ))
+            }
+        }
+
+        net = InternalMockNetwork(
+                cordappPackages = listOf("com.luxoft.blockchainlab.corda.hyperledger.indy"),
+                networkParameters = testNetworkParameters(maxTransactionSize = 10485760 * 5))
+
+        parties.clear()
+    }
+
+    @After
+    fun commonTearDown() {
+        try {
+            for (party in parties) {
+                for(service in projectServises) {
```

Comment:
IndyService is just one service who uses IndyClient and has close function. 
If you will extend projectServices it breaks the logic. I suggest don't to use such construction and close the wallets in IndyService directly

Reply:
Good point, will fix